src/MfaVerifyController.js

```diff
@@ -118,11 +118,10 @@ function (Okta, Checkbox, BaseLoginController, CookieUtil, TOTPForm, YubikeyForm
     },
 
     trapAuthResponse: function () {
-      if(this.options.factorType === 'password') {
-        return false;
-      }
-      if (this.options.appState.get('isMfaChallenge') ||
-          this.options.appState.get('isMfaRequired')) {
+      if((this.options.appState.get('isMfaRequired') &&
+          this.options.appState.get('trapMfaRequiredResponse')) ||
+          this.options.appState.get('isMfaChallenge')) {
+        this.options.appState.set('trapMfaRequiredResponse', false);
```

Comment:
I would like to understand this a little more... so before if we were in MFA_REQUIRED or MFA_CHALLENGE, we `trap` the response, meaning we don't let ```RouterUtil``` navigate somewhere else.

With this change, we will trap the reponse when we are either in MFA_CHALLENGE or we are in MFA_REQUIRED and we explicitly set a variable to trap it. This means that if we don't set the variable, we will let RouterUtil navigate to the next view. Is this correct? If so, then my next question is for your changes in ```InlineTOTPForm.js```, why is it setting ```trapMfaRequiredResponse``` to true?

And well, I think in general the idea is (correct me if I am missing something)
- There are 2 flows that come here: from the factors dropdown or from an api response.
- If we interact with the dropdown, we should always trap the response, no matter what status we are in.
- If we get a response that says MFA_REQUIRED, then we should always navigate to a new view (with the exception of the above item involving a dropdown)

Reply:
Spoke to Mau about this offline. The general idea above is correct. Made changes to set trapMfaRequiredResponse only when moving from MFA_CHALLENGE to MFA_REQUIRED state